src/libstore/store-api.hh

```diff
@@ -635,6 +635,9 @@ public:
         return 0;
     };
 
+    /* Check if the user making the connection is a trusted-user. */
+    virtual bool isTrustedUser() = 0;
```

Comment:
I think it's fine to define this to `unsupported` by default like we do for a bunch of other methods and remove all the custom implementations for the stores that don't support it:

```suggestion
    virtual bool isTrustedUser()
    { unsupported("isTrustedUser"); }
```

Reply:
That definitely makes it a smaller diff too.

Reply:
Done. I've left the implementation for legacy ssh though, since it's more helpful to print it.

Reply:
@roberth since the default behaviour is to fail loudly, I think it's OK because it's a conservative approach, and we can always refine the implementation after the fact if we realize that we've been missing one